lib/faraday_middleware/response/follow_redirects.rb

```diff
@@ -130,5 +143,22 @@ def safe_escape(uri)
         '%' + match.unpack('H2' * match.bytesize).join('%').upcase
       }
     end
+
+    def clear_authorization_header(env, from_url, to_url)
+      return env if redirect_to_same_host?(from_url, to_url)
+      return env unless @options.fetch(:clear_authorization_header, true)
+
+      env[:request_headers].delete(AUTH_HEADER)
+      env
```

Comment:
No need to return `env` (same for the 2 returns above).
See previous comment for explanation.